free-1.md

```diff
@@ -10,41 +10,148 @@ _What need is met by your idea? Who are you helping? What is that person's
+input and produces a single sound as output.
+
+The second part of programming is writing a score. A score uses various
+instruments to produce a piece of music by playing a certain note on a certain
+instrument at a certain time. Playing a note means giving the instrument any
+inputs that it requires to start generating sound, such as pitch and amplitude.
+There's some flexibility here: the score-creation process could be a textual
+language that comes bundled with the instrument-creation language. This is what
+CSound does. Or, the instrument language could be designed to interface with
+existing methods for writing musical scores. For example, there could be a way
+to assign each track in a MIDI file to a different instrument, and use the MIDI
+messages like note-on, note-off, velocity and so on to pass parameters to the
+instruments.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+The output of the program could be either a sound file or an actual sound.
+Either way, the runtime looks the same; the only difference is whether samples
+are written to a file or to the computers DAC. The basic runtime process
+consists of reading "notes" from the score in chronological order, and calling
+the correct instrument for each note. When the instrument is called, it starts
+generating a signal from the first source in its chain. That signal is then
+passed as the input to the next component of the chain, and so on until the end
+of the chain. The resulting signal is written to the output.
+
+There could be errors in the score, such as passing the wrong parameters to an
+instrument trying to call an unknown instrument. A simple error message
+describing where in the score the error occurred and what it was would probably
+be enough to communicate to the user. In the instrument definitions, possible
+errors include ill-typed connections. For example, trying to connect the output
+of a source to another source, which does not use a signal as input, would be
+an error. These could also be communicated as error messages.
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It should be easy to create instruments that are based on commonly-used patterns like frequency modulation or physical modeling. It should be harder, but possible, to create completely different kinds of instruments. This may mean exposing CSound-like primitives, but encouraging users to use the higher level constructs where possible.
+
+It should be difficult or almost impossible to produce a complete song with this language. The language is not meant to be used for recording, mixing or mastering. More likely, the user would need to combine instruments written in this language with a fully-feature DAW to produce a complete song.
```

Comment:
I agree that it should be difficult to write a complete song. That would require you to limit the ability to write a score, which you want. It would be interesting to see how you handle this situation.